js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+    this._artboard.advance(0);
+    this._artboard.draw(this.renderer);
+    this.ctx.restore();
+  }
+
+  // Adds animations contained in the artboard for playback
+  private playAnimations(animationNames?: string | string[]): string[] {
+    animationNames = mapToStringArray(animationNames);
+    const instancedAnimationNames = this._animations.map(a => a.name);
+    for (const i in animationNames) {
+      const index = instancedAnimationNames.indexOf(animationNames[i]);
+      if (index >= 0) {
+        // Animation is already instanced, unpause it
+        this._animations[index].paused = false;
+      } else {
+        // Create a new animation instance and add it to the list
+        const anim = this._artboard.animation(animationNames[i]);
+        const inst = new this._rive.LinearAnimationInstance(anim);
+        this._animations.push(new Animation(anim, inst));
+      }
+    }
+  
+    return this._animations.filter(a => !a.paused).map(a => a.name);
+  }
+
+  // Removes animations from playback
+  private removeAnimations(animationNames: string[]): string[] {
+    // Determine which animations need to be removed
+    const animationsToRemove = this._animations.filter(
+      a => animationNames.indexOf(a.name) >= 0
+    );
+
+    // Remove the animations
+    animationsToRemove.forEach(a =>
+      this._animations.splice(this._animations.indexOf(a), 1)
+    );
+
+    // Return the list of animations removed
+    return animationsToRemove.map(a => a.name);
+  }
+
+  // Removes all animations from playback
+  private removeAllAnimations(): string[] {
+    const names = this._animations.map(animation => animation.name);
+    this._animations.splice(0, this._animations.length);
+    return names;
+  }
+
+  // Pauses animations
+  private pauseAnimations(animationNames: string[]): string[] {
+    const pausedAnimationNames: string[] = [];
+
+    this._animations.forEach((a, i) => {
+      if (animationNames.indexOf(a.name) >= 0) {
+        a.paused = true;
+        pausedAnimationNames.push(a.name);
+      }
+    });
+    return pausedAnimationNames;
+  }
+
+  // Returns true if at least one animation is active
+  private hasPlayingAnimations(): boolean {
+    return this._animations.reduce((acc, curr) => acc || !curr.paused, false);
+  }
+
+  // Ensure there's at least one animation for playback; if there are none
+  // marked for playback, then ad the first animation in the artboard.
+  private atLeastOneAnimationForPlayback(): void {
+    if (this._animations.length === 0 && this._artboard.animationCount() > 0) {
+      // Add the default animation
+      const animation = this._artboard.animationAt(0);
+      const instance = new this._rive.LinearAnimationInstance(animation);
+      this._animations.push(new Animation(animation, instance));
+    }
+  }
+
+  // Tracks the last timestamp at which the animation was rendered. Used only in
+  // draw().
+  private lastRenderTime: number;
```

Comment:
This is consider a good practice to define all the properties before the constructor

Reply:
Left-over from porting the js; I'll tidy that up.